ocf_data_sampler/load/satellite.py

```diff
@@ -9,51 +16,203 @@
+            else:
+                raise ValueError(f"Expected snapshot {sha1}, but main branch has {session.snapshot_id}")
+                
+        except Exception as e:
+            # NO FALLBACK - error out as Sol requested
+            raise ValueError(f"Commit {sha1} not found or inaccessible: {e}")
+    else:
+        logger.info("Opening 'main' branch of Ice Chunk repository.")
+        session = repo.readonly_session("main")
 
-    check_time_unique_increasing(ds.time_utc)
+    # Open the dataset from the Ice Chunk session store
+    ds = xr.open_zarr(session.store, consolidated=True, chunks="auto")
+
+    # Convert Ice Chunk format to standard format
+    if len(ds.data_vars) > 1:
+        # Data is stored with separate variables per channel - combine them
+        data_arrays = [ds[var] for var in sorted(ds.data_vars)]
+        combined_da = xr.concat(data_arrays, dim="variable")
+        combined_da = combined_da.assign_coords(variable=sorted(ds.data_vars))
+        
+        # Create dataset with data variable (to match OCF format)
+        ds = xr.Dataset({"data": combined_da})
+
+    return ds
+
+def get_single_sat_data(zarr_path: str) -> xr.Dataset:
+    """Opens a single satellite zarr file with Ice Chunk support via match/case patterns.
+    
+    This function implements Sol's requested architecture consolidating all conditional 
+    logic into match/case patterns for different path types.
+    """
+    # Parse path components using Sol's regex approach
+    path_info = _parse_zarr_path(zarr_path)
+    
+    # Sol's requested match/case pattern for path routing
+    match path_info:
+        case {"protocol": protocol, "bucket": bucket, "prefix": prefix, "sha1": sha1} if ".icechunk" in prefix and protocol is not None:
+            # Ice Chunk logic goes here - Sol's requested signature
+            ds = _open_sat_data_icechunk(protocol, bucket, prefix, sha1)
+        
+        case {"protocol": _, "bucket": _, "prefix": _, "sha1": None}:
+            # Existing single zarr logic path here - Sol's specification
+            ds = open_zarr(zarr_path)
+        
+        case _:
+            # Raise error on unhandled path
+            raise ValueError(f"Unhandled path format: {zarr_path}")
+    
+    return ds
+
+def open_sat_data(zarr_path: str | list[str], channels: Optional[List[str]] = None) -> xr.DataArray:
+    """Lazily opens the zarr store and validates data types.
+    
+    This function returns to "as-was" as requested by Sol, with all conditional 
+    logic moved to get_single_sat_data.
+    """
+    if isinstance(zarr_path, list):
+        # Handle multiple zarr files
+        ds_list = []
+        for path in zarr_path:
+            ds_list.append(get_single_sat_data(path))
+        
+        ds = xr.combine_nested(
+            ds_list,
+            concat_dim="time",
+            combine_attrs="override",
+            join="override",
+        )
```

Comment:
Because icechunks are single folders for the full store, I think we can assume that we won't ever want to open more than one at once for a single data source - and as such this conditional branch can return to the original one, with `open_zarrs` imported from `.open_tensorstore_zarrs`:

```
if isinstance(zarr_path, list | tuple):
        ds = open_zarrs(zarr_path, concat_dim="time")
```